drivers/soundwire/bus.c

```diff
@@ -753,6 +1103,46 @@ static int sdw_initialize_slave(struct sdw_slave *slave)
 	return 0;
 }
 
+static int sdw_deprep_clk_stop1(struct sdw_slave *slave)
+{
+	int ret;
+
+	/*  Check property to deprepare the Slave */
+	if (!slave->prop.reset_behave)
+		return 0;
+
+	ret = sdw_read_no_pm(slave, SDW_SCP_SYSTEMCTRL);
+	if (ret <  0) {
+		dev_err(slave->bus->dev,
+			"SDW_SCP_SYSTEMCTRL read failed:%d", ret);
+		return ret;
+	}
+
+	ret = sdw_slave_clk_stop_callback(slave, SDW_CLK_STOP_MODE1,
+					  SDW_CLK_PRE_DEPREPARE);
+
+	if (ret < 0)
+		return ret;
+
+	ret = sdw_slave_clk_stop_prepare(slave, SDW_CLK_STOP_MODE1, false);
+	if (ret < 0)
+		return ret;
+
+	/* Wait till de-prepare is complete by checking NF bit */
+	if (!slave->prop.simple_clk_stop_capable) {
+		ret = sdw_bus_wait_for_clk_prep_deprep(slave->bus,
+						       slave->dev_num);
+		if (ret < 0)
+			return ret;
+	}
+
+	/* Inform driver */
+	ret = sdw_slave_clk_stop_callback(slave, SDW_CLK_STOP_MODE1,
+					  SDW_CLK_POST_DEPREPARE);
+
+	return ret;
+}
+
```

Comment:
let's drop clock_stop_mode1, it's not really useful

Reply:
ok, I will refine it